fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -1338,22 +1342,29 @@ public boolean shouldAllowUniquePendingState(FDBRecordStore store) {
          */
         public boolean shouldAllowTypeConversionContinue(IndexBuildProto.IndexBuildIndexingStamp.Method newMethod,
                                                          IndexBuildProto.IndexBuildIndexingStamp.Method oldMethod) {
-            if (allowedTakeoverSet != null) {
-                if (newMethod == IndexBuildProto.IndexBuildIndexingStamp.Method.BY_RECORDS) {
-                    if (oldMethod == IndexBuildProto.IndexBuildIndexingStamp.Method.MULTI_TARGET_BY_RECORDS) {
-                        return allowedTakeoverSet.contains(TakeoverTypes.MULTI_TARGET_TO_SINGLE);
-                    }
-                    if (oldMethod == IndexBuildProto.IndexBuildIndexingStamp.Method.MUTUAL_BY_RECORDS) {
-                        return allowedTakeoverSet.contains(TakeoverTypes.MUTUAL_TO_SINGLE);
-                    }
+            if (newMethod == IndexBuildProto.IndexBuildIndexingStamp.Method.BY_RECORDS) {
+                if (oldMethod == IndexBuildProto.IndexBuildIndexingStamp.Method.MULTI_TARGET_BY_RECORDS) {
+                    return isTypeConversionAllowed(TakeoverTypes.MULTI_TARGET_TO_SINGLE);
+                }
+                if (oldMethod == IndexBuildProto.IndexBuildIndexingStamp.Method.MUTUAL_BY_RECORDS) {
+                    return isTypeConversionAllowed(TakeoverTypes.MUTUAL_TO_SINGLE);
```

Comment:
I know this existed before, but the comment below makes me wonder what happens if you have a `MUTUAL_BY_RECORDS` that has multiple targets? Does it get split up?

Reply:
Yes. converting to `BY_RECORDS` will split any multi target. The original motivation was a fallback. 
I will add a comment to clarify.

Reply:
✅